### PR TITLE
Added IFDEFs to make JCL compatible with FPC 64-bit

### PR DESCRIPTION
**Pull request by denistotoliciu**
Here is a set of small modifications mostly related to conditional compilation in order to make the JCL package compilable by the FPC 64-bit compiler.

### CODE REVIEW

File: jcl/source/common/JclRTTI.pas [hunk 1]
```diff
@@ -2223,12 +2223,12 @@ function TJclDynArrayTypeInfo.GetElementType: IJclTypeInfo;
...
   begin
     if TypeData.elType2 <> nil then
-      Result := JclTypeInfo(TypeData.elType2^)
+      Result := JclTypeInfo(TypeData.elType2{$IFDEF DELPHICOMPILER}^{$ENDIF})
     else
       Result := nil;
   end
   else
-    Result := JclTypeInfo(TypeData.elType^);
+    Result := JclTypeInfo(TypeData.elType{$IFDEF DELPHICOMPILER}^{$ENDIF});
```
Comment (obones) [inline jcl/source/common/JclRTTI.pas:2231]: Same as above

File: jcl/source/common/JclSimpleXml.pas [hunk 2]
```diff
@@ -629,7 +629,7 @@   TXMLVariant = class(TInvokeableVariantType)
       const Name: string; const Arguments: TVarDataArray): Boolean; override;
     function GetProperty(var Dest: TVarData; const V: TVarData;
       const Name: string): Boolean; override;
-    function SetProperty(const V: TVarData; const Name: string;
+    function SetProperty(var V: TVarData; const Name: string;
```
Comment (obones) [inline jcl/source/common/JclSimpleXml.pas:632]: Why change this to `var`? The `SetProperty` method does not change the value of `V` so there is no reason to make it a `var` parameter

File: jcl/source/common/JclSimpleXml.pas [hunk 3]
```diff
@@ -4244,7 +4244,7 @@ function TXMLVariant.IsClear(const V: TVarData): Boolean;
   Result := (VXML = nil) or (not VXML.HasItems);
 end;
 
-function TXMLVariant.SetProperty(const V: TVarData; const Name: string;
+function TXMLVariant.SetProperty(var V: TVarData; const Name: string;
```
Comment (obones) [inline jcl/source/common/JclSimpleXml.pas:4247]: See above

File: jcl/source/common/JclStrings.pas [hunk 4]
```diff
@@ -5390,12 +5390,12 @@ function CompareNatural(const S1, S2: string; CaseInsensitive: Boolean): SizeInt
   end;
 end;
 
-function CompareNaturalStr(const S1, S2: string): SizeInt; overload;
+function CompareNaturalStr(const S1, S2: string): SizeInt;{$IFDEF DELPHICOMPILER}overload;{$ENDIF}
```
Comment (obones) [inline jcl/source/common/JclStrings.pas:5393]: Same as in previous files, FPC 2.6 does not complain here

File: jcl/source/common/JclSysInfo.pas [hunk 5]
```diff
@@ -4191,8 +4191,6 @@ function GetWindows10ReleaseName: String;
           Result := 'Windows 10 October 2018 Update';
        1903:
           Result := 'Windows 10 May 2019 Update';
-       1909:
-          Result := 'Windows 10 November 2019 Update';
```
Comment (obones) [inline jcl/source/common/JclSysInfo.pas:4195]: Why remove those lines? This Windows version did not disappear...

File: jcl/source/common/JclSysInfo.pas [hunk 6]
```diff
@@ -4224,8 +4222,6 @@ function GetWindows10ReleaseCodeName: String;
           Result := 'Redstone 5';
        1903:
           Result := '19H1';
-       1909:
-          Result := '19H2';
```
Comment (obones) [inline jcl/source/common/JclSysInfo.pas:4228]: Same as above

File: jcl/source/windows/JclCOM.pas [hunk 7]
```diff
@@ -486,7 +486,7 @@ function UnRegisterCLSIDInCategory(const ClassID: TGUID; const CatID: TGUID): HR
 
 function ResetIStreamToStart(Stream: IStream): Boolean;
 var
-  i64Pos: {$IFDEF RTL290_UP}LargeUInt{$ELSE}Largeint{$ENDIF RTL290_UP};
+  i64Pos: {$IF DEFINED(RTL290_UP) OR DEFINED(CPU64)}LargeUInt{$ELSE}Largeint{$ENDIF RTL290_UP};
```
Comment (obones) [inline jcl/source/windows/JclCOM.pas:489]: I don't see how a 64bits CPU forces to use the unsigned type

File: jcl/source/windows/JclRegistry.pas [hunk 8]
```diff
@@ -398,7 +402,7 @@ implementation
   Winapi.AccCtrl,
   {$ELSE ~HAS_UNITSCOPE}
   SysUtils,
-  AccCtrl,
+  {$IFDEF DELPHILANGUAGE}AccCtrl,{$ENDIF ~DELPHILANGUAGE}
```
Comment (obones) [inline jcl/source/windows/JclRegistry.pas:405]: Delphi does not define `DELPHILANGUAGE` so basically, you removed `AccCtrl` for all Delphi versions. And the `~` in the `ENDIF` is specifically for when using `IFNDEF` which is not the case here.

File: jcl/source/windows/JclStructStorage.pas [hunk 9]
```diff
@@ -730,7 +730,7 @@ function TJclStructStorageStream.Clone: TJclStructStorageStream;
 function TJclStructStorageStream.CopyTo(Stream: TJclStructStorageStream;
   Size: Int64): Boolean;
 var
-  DidRead, DidWrite: {$IFDEF RTL290_UP}LargeUInt{$ELSE}Largeint{$ENDIF RTL290_UP};
+  DidRead, DidWrite: {$IF DEFINED(RTL290_UP) OR DEFINED(CPU64)}LargeUInt{$ELSE}Largeint{$ENDIF RTL290_UP};
```
Comment (obones) [inline jcl/source/windows/JclStructStorage.pas:733]: Same as above, I don't see why a 64bits CPU forces to use an unsigned integer

File: jcl/source/windows/JclStructStorage.pas [hunk 10]
```diff
@@ -774,7 +774,7 @@ function TJclStructStorageStream.Read(var Buffer; Count: Longint): Longint;
 
 function TJclStructStorageStream.Seek(Offset: Integer; Origin: Word): Longint;
 var
-  N: {$IFDEF RTL290_UP}LargeUInt{$ELSE}Largeint{$ENDIF RTL290_UP};
+  N: {$IF DEFINED(RTL290_UP) OR DEFINED(CPU64)}LargeUInt{$ELSE}Largeint{$ENDIF RTL290_UP};
```
Comment (obones) [inline jcl/source/windows/JclStructStorage.pas:777]: Same as above

File: jcl/source/windows/JclWin32.pas [hunk 11]
```diff
@@ -55,7 +55,11 @@
 {$I windowsonly.inc}
 
 {$MINENUMSIZE 4}
-{$ALIGN ON}
+{$IFDEF WIN64}
+  {$ALIGN 8}
+{$ELSE}
+  {$ALIGN 4}
+{$ENDIF}
```
Comment (obones) [inline jcl/source/windows/JclWin32.pas:62]: Why force an alignment that is not the default one from the compiler, which by the way is different for the CPU it targets.

File: jcl/packages/fpc/Jcl.lpk [hunk 12]
```diff
@@ -1,17 +1,16 @@
-<?xml version="1.0"?>
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (denistotoliciu) [reply jcl/packages/fpc/Jcl.lpk:1]: Sorry, my bad, the IDE updated some of its fields automatically. Will use the previous version from Git.

**Comment by denistotoliciu** [comment]
Will make a new pull request with additional commits that will make more things clearer.

**Pull request closed without merging**